storage/store/payloads.go

```diff
@@ -0,0 +1,165 @@
+package store
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+type Payloads struct {
+	db         storage.DB
+	index      *Index
+	guarantees *Guarantees
+	seals      *Seals
+	receipts   *ExecutionReceipts
+	results    *ExecutionResults
+}
+
+func NewPayloads(db storage.DB, index *Index, guarantees *Guarantees, seals *Seals, receipts *ExecutionReceipts,
+	results *ExecutionResults) *Payloads {
+
+	p := &Payloads{
+		db:         db,
+		index:      index,
+		guarantees: guarantees,
+		seals:      seals,
+		receipts:   receipts,
+		results:    results,
+	}
+
+	return p
+}
+
+// TODO(leo): rename to BatchStore, add to interface
+func (p *Payloads) storeTx(rw storage.ReaderBatchWriter, blockID flow.Identifier, payload *flow.Payload, storingResults map[flow.Identifier]*flow.ExecutionResult) error {
+	// For correct payloads, the execution result is part of the payload or it's already stored
+	// in storage. If execution result is not present in either of those places, we error.
+	// ATTENTION: this is unnecessarily complex if we have execution receipt which points an execution result
+	// which is not included in current payload but was incorporated in one of previous blocks.
```

Comment:
I don't understand why you are stating "is _unnecessarily_ complex". Even if the result was incorporated in one of previous blocks, it would still make sense to sanity check, right? I would suggest to either remove this statement entirely, or rework the explanation.

Reply:
Yeah, I will just remove it.